File: sprint04/KDH/02/PGS_42626.py
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)

    while len(scoville) >= 2:
        if scoville[0] >= K:
            return answer
        min1 = heapq.heappop(scoville)
        min2 = heapq.heappop(scoville)

        heapq.heappush(scoville, min1+2*min2)
        answer += 1

    return answer if scoville[0] >= K else -1

File: sprint04/JMS/SW/PGS_42626.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> scoville, int K) {
    multiset<int> s;
    int answer = 0;
    
    for(auto p : scoville)
        s.insert(p);
    
    while(1){
        int sum=0;
        
        sum += *s.begin();
        s.erase(s.begin());
        
        if(sum >= K)
            return answer;
        
        if(s.size() == 0)
            return -1;
        
        sum = sum + *s.begin() * 2;
        answer++;
        s.erase(s.begin());
        s.insert(sum);
    }
}
